lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeTest.kt

```diff
@@ -0,0 +1,788 @@
+                ),
+                InvalidTestCase(
+                    "dateTo",
+                    "dateTo=2020-01-01",
+                    "Metadata field dateTo does not exist",
+                ),
+                InvalidTestCase(
+                    "intFieldFrom",
+                    "intFieldFrom=1",
+                    "Metadata field intFieldFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "intFieldTo",
+                    "intFieldTo=1",
+                    "Metadata field intFieldTo does not exist",
+                ),
+                InvalidTestCase(
+                    "floatFieldFrom",
+                    "floatFieldFrom=1",
+                    "Metadata field floatFieldFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "floatFieldTo",
+                    "floatFieldTo=1",
+                    "Metadata field floatFieldTo does not exist",
+                ),
+            ).map { arrayOf(it.description, it.query, it.expected) }
+    }
+
+    @ParameterizedTest(name = "valid {0} with >= and <=")
+    @MethodSource("validRangeQueryProvider")
+    fun `test valid advanced queries`(
+        description: String,
+        query: String,
+        expected: SiloFilterExpression,
+    ) {
+        val result = underTest.map(query)
+        assertThat(result, equalTo(expected))
+    }
+
+    @ParameterizedTest(name = "invalid {0} with >= and <=")
+    @MethodSource("invalidRangeQueryProvider")
+    fun `test invalid advanced queries`(
+        description: String,
+        query: String,
+        expected: String,
+    ) {
+        val exception = assertThrows<BadRequestException> { underTest.map(query) }
+        assertThat(exception.message, containsString(expected))
+    }
+
+    @ParameterizedTest(name = "invalid metadata field {0}")
+    @MethodSource("invalidMetadataProvider")
+    fun `given a advancedQuery with xFrom or xTo field throw BadRequestException`(
+        description: String,
+        query: String,
+        expected: String,
+    ) {
+        val exception =
+            assertThrows<BadRequestException> { underTest.map(query) }
+        assertThat(
+            exception.message,
+            `is`(
+                expected,
+            ),
+        )
+    }
+
+    @Test
+    fun `given a advancedQuery with not allowed fields in IsNull throw BadRequestException`() {
+        val advancedQueries =
+            listOf(
+                "dateFrom=2020-01-01",
+                "dateTo=2020-01-01",
+                "intFieldFrom=1",
+                "intFieldTo=1",
+                "floatFieldFrom=1",
+                "floatFieldTo=1",
+                "metadata.regex=1",
+            )
+        for (advancedQuery in advancedQueries) {
+            val query = "IsNull(%s)".format(advancedQuery)
+            val exception = assertThrows<BadRequestException> { underTest.map(query) }
```

Comment:
`exception` is unused, we should check the message.

But actually I would prefer to merge those with the other invalid tests and use a parametrized test instead.